File: big_data_learning/spark-learning/com/shawn/study/spark/word-count.py
from pyspark import SparkConf, SparkContext

if __name__ == '__main__':
    conf = SparkConf().setMaster("local").setAppName('spark-learning')
    sc = SparkContext(conf=conf)
    inputPath = "F:\\dev\\learning-notes\\big_data_learning\\spark-learning\\resources\\quickstart\\work-count"
    # create RDD from a text file
    baseRDD = sc.textFile(inputPath)
    print(baseRDD.collect())

    wordsRDD = baseRDD.flatMap(lambda line: line.split(" "))
    print(wordsRDD.collect())

    pairsRDD = wordsRDD.map(lambda word: (word, 1))
    print(pairsRDD.collect())

    frequenciesRDD = pairsRDD.reduceByKey(lambda a, b: a + b)
    print(frequenciesRDD.collect())

    # done!
    sc.stop()

File: big_data_learning/spark-scala-java-learning/src/main/java/com/shawn/study/big/data/spark/scala/core/quickstart/WordCount.scala
package com.shawn.study.big.data.spark.scala.core.quickstart

import org.apache.spark.{SparkConf, SparkContext}

object WordCount {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("spark-scala-learning").setMaster("local")
    val context = new SparkContext(conf)
    val path = "examples/input/wordcount"
    context.textFile(path).flatMap(_.split(" "))
      .map((_, 1))
      .reduceByKey(_ + _).collect().foreach(println)
  }
}
